src/Sylius/Bundle/PayumBundle/Extension/UpdatePaymentStateExtension.php

```diff
@@ -55,8 +55,16 @@ public function onExecute(Context $context)
      */
     public function onPostExecute(Context $context)
     {
-        if ($context->getPrevious()) {
+        $previousStack = $context->getPrevious();
+        $stackSize = count($previousStack);
+        
+        if ($stackSize > 1) {
             return;
+        } elseif ($stackSize === 1) {
+            $previousActionClassName = get_class($previousStack[0]->getAction());
+            if (false === stripos($previousActionClassName, 'NotifyNullAction')) {
```

Comment:
@pamil Because this is gateway-specific implementation and Payum does not have a special interface for these - for every gateway that has to implement this, it will sit inside that gateway's extension namespace. That's why we can't use instanceof - those will be different types all the time.